doc/1.4/language.md

```diff
@@ -1942,6 +1942,29 @@ hook(int *, bool) h3;
 Beyond suspending computations on it, a hook <tt><em>h</em></tt> has two
 associated operations:
 
+* <pre><em>h</em>.send(<em>msg1</em>, ... <em>msgN</em>)</pre>
+  Sends a message through the hook, with message components
+  <tt><em>msg1</em></tt> through <tt><em>msgN</em></tt>. The number of message
+  components must match the number of message component types of the hook,
+  and each message component must be compatible with the corresponding message
+  component type of the hook.
+
+  `send` is *asynchronous*: the message will only be sent &mdash; and suspended
+  computations executed &mdash; once all current device entries on the call
+  stack have been completed. It is exactly equivalent to
+  <tt>after: <em>h</em>.send_now(<em>msg1</em>, ... <em>msgN</em>)</tt>, except
+  it's not possible to prevent the message from being sent via `cancel_after()`.
+  For more information, see [Immediate After
+  Statements](#immediate-after-statements).
+
+  Like immediate after statements, pointers to stack-allocated data **must not**
+  be passed as message components to a `send`. If you must use pointers to
+  stack-allocated data, then `send_now` should be used instead of `send`. If you
+  want the message to be delayed to avoid ordering bugs, create a method which
+  wraps the `send_now` call together with the declarations of the local
+  variable(s) which are pointed to, and then use immediate after (`after:
```

Comment:
"an immediate after statement", grammatically problematic to use "after" as a noun. This is the only end-user visible occurrence I could spot, I won't oppose the identifiers.